src/java.base/share/classes/sun/security/ssl/SSLCipher.java

```diff
@@ -1859,10 +1859,20 @@ public Plaintext decrypt(byte contentType, ByteBuffer bb,
                 }
 
                 if (bb.remaining() <= tagSize) {
-                    throw new BadPaddingException(
-                        "Insufficient buffer remaining for AEAD cipher " +
-                        "fragment (" + bb.remaining() + "). Needs to be " +
-                        "more than tag size (" + tagSize + ")");
+                    // Check for unexpected plaintext alert.
+                    if (contentType == ContentType.ALERT.id
+                        && bb.remaining() == 2) {
+                        throw new GeneralSecurityException(String.format(
+                            "Unexpected plaintext alert received: " +
+                            "Level: %s; Alert: %s",
+                            Alert.Level.nameOf(bb.get(bb.position())),
+                            Alert.nameOf(bb.get(bb.position() + 1))));
+                    } else {
+                        throw new BadPaddingException(
+                            "Insufficient buffer remaining for AEAD cipher " +
+                            "fragment (" + bb.remaining() + "). Needs to be " +
+                            "more than tag size (" + tagSize + ")");
+                    }
```

Comment:
From the context, we only know the record is an alert message.  We don't actually know it is plaintext or not.  But I will have you for the final decision.

There are another "Insufficient buffer remaining" exception in the SSLCipher class, we may want to keep the behavior consistent, with similar update.  But I am fine to do it separately.

I am going to approve it.

Reply:
- I'm not sure what you mean by `plaintext`, what is your definition of plaintext? Per TLS1.3 RFC it's a plaintext unless  contentType == 23
- About other "Insufficient buffer remaining" exceptions: from what we know so far this problem is specific to TLSv1.3, so I think we'll deal with other cases separately if needed in the future.

Reply:
> * I'm not sure what you mean by `plaintext`, what is your definition of plaintext? Per TLS1.3 RFC it's a plaintext unless  contentType == 23

With "plaintext", I meant it is not TLSCiphertext.   You have a good point.  Thank you!